src/image_store/mem.rs

```diff
@@ -53,6 +53,16 @@ impl Store {
     pub fn remove(&mut self, filename: &str) -> Option<File> {
         self.files.remove(filename)
     }
+
+    pub fn remove_by_prefix(&mut self, filename_prefix: &str) -> Option<(Box<str>, File)> {
+        // find key with prefix
+        let key = self.files.keys()
+            .find(|key| key.starts_with(filename_prefix))
+            .cloned();
+
+        // if key found, remove entry and return file + filename
+        key.and_then(|k| self.files.remove(&k).map(|file| (k, file)))
```

Comment:
```suggestion
        // find key with prefix
        let key = self.files.keys()
            .find(|key| key.starts_with(filename_prefix))
            .cloned();

        // if key found, remove entry and return key + file
        key.and_then(|ref k| self.files.remove_entry(k))
```
There is a remove_entry function on HashMap that does the map stuff internally. Also, ref does a reference capture from closure.